topo/node/ceos/ceos.go

```diff
@@ -171,18 +163,16 @@ func (n *Node) GenerateSelfSigned(ctx context.Context, ni node.Interface) error
 		),
 	}
 
-	r, err := n.cliConn.SendConfigs(cfgs)
+	resp, err := n.cliConn.SendConfigs(cfgs)
```

Comment:
still haven't removed the "multiple error" pattern i thought that was a goal of this pr?

Reply:
the goals as I understood it were to fix the transport interfaces up so that the testing stuff didn't leak into production bits as well as to make `Failed` return an error (or nil ofc) instead of a bool. that `Failed` error also now contains the contextual info about the "error" that was encountered.

the long standing scrapli pattern here is that we only raise exception (in python ofc) or return an error if the connection encounters an error, so unless that happens we always return a response. if we get that response we have that extra feature to test if the remote device produced some application error (ie invalid command/wrong port/whatever). so basically the api contract w/in scrapli is to *not* error out if the inputs were delivered properly and outputs read off the wire properly.

we can remove the extra "level of checking" here if you prefer but this pattern has been working quite well for scrapli so the idea was brought into go version and felt useful/valuable here too.